pkg/handlers/v1/nexposeassetproducer_test.go

```diff
@@ -43,36 +44,42 @@ func TestNexposeAssetProducerHandler(t *testing.T) {
 	handler.Handle(context.Background(), scanInfo)
 }
 
+// This test not only handles testing multiple incoming assets, but also
+// tests whether we have race conditions. The race conditions are related to
+// the metric we want to keep track of and emit at the end of the Handle function
 func TestNexposeAssetProducerHandlerMultipleAssets(t *testing.T) {
+	const numberOfGoRoutines = 1000000
```

Comment:
nit: This number is a little excessive. This will case the channel creation below to allocate memory for 1,000,000 `domain.AssetEvent` types. Even if these were 32bit integer pointers we'd end up with at least 4MiB of allocation. It's likely this results in our test allocating double digit MiB for the tests. Not really a big problem but also not really the size we _need_.